backend/de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/ad/trx/api/impl/TrxListenerManager.java

```diff
@@ -96,16 +80,22 @@ private void verifyEventTiming(@NonNull final RegisterListenerRequest listener)
 		final TrxEventTiming eventTimingOfListener = listener.getTiming();
 
 		final TrxEventTiming currentTiming = getCurrentTiming();
+
+		if (currentTiming == eventTimingOfListener && listener.isForceAfterNextCommit())
```

Comment:
i don't really understand this boolean expression; maybe you can try making it clearer by extracking local vars or by adding a comment?